test/boost/bloom_filter_test.cc

```diff
@@ -96,9 +96,9 @@ SEASTAR_TEST_CASE(test_sstable_manager_auto_reclaim_and_reload_of_bloom_filter)
         // Test auto reload - disposing sst3 should trigger reload of the
         // smallest filter in the reclaimed list, which is sst1's bloom filter.
         dispose_and_stop_tracking_bf_memory(std::move(sst3), sst_mgr);
-        REQUIRE_EVENTUALLY_EQUAL(sst1->filter_memory_size(), sst1_bf_memory);
+        REQUIRE_EVENTUALLY_EQUAL<uint64_t>([&] { return sst1->filter_memory_size(); }, sst1_bf_memory);
```

Comment:
Why is explicit instantiation needed, isn't the compiler able to deduce the type?

Reply:
Unfortunately not.
```
test/boost/bloom_filter_test.cc:99:9: error: no matching function for call to 'REQUIRE_EVENTUALLY_EQUAL'
   99 |         REQUIRE_EVENTUALLY_EQUAL([&] { return sst1->filter_memory_size(); }, sst1_bf_memory);
      |         ^~~~~~~~~~~~~~~~~~~~~~~~
./test/lib/eventually.hh:77:6: note: candidate template ignored: could not match 'std::function<T ()>' against '(lambda at test/boost/bloom_filter_test.cc:99:34)'
   77 | void REQUIRE_EVENTUALLY_EQUAL(std::function<T()> a, T b, sleep_fn sleep = seastar_sleep_fn{}) {
      |      ^
1 error generated.
```